Modules/events/event.py

```diff
@@ -0,0 +1,181 @@
+"""
+event.py - Event system
+
+A Class decorator event system.
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
+import logging
+from functools import partial
+from typing import Callable, List, Dict, Any, Union
+
+# typedef
+Subscriptions = List[Callable]
+log = logging.getLogger(f"mecha.{__name__}")
+
+# If a subscriber returns this value, the event will canceled.
+# return this value to consume an event
+CANCEL_EVENT = object()
+
+
+class Event:
+    """
+        Defines an event.
+
+        an event is a subscribe-able object that will call its subscribers when it is invoked.
+
+    Examples:
+
+         an event can also be defined without a function definition
+         >>> my_other_event = Event("my_other_event")
+         >>> "my_other_event" in Event.events
+         True
+
+    Notes:
+        - Events will need to be invoked by something.
+
+        - Event subscribers must be async functions, and should be tested to work with
+            the event's signature (variable)
+
+        - exceptions raised by subscribers will not stop an event from being emitted
```

Comment:
Wasn't the event already emitted when a subscriber has the chance to throw an exception? In that case, maybe 'will not stop other subscribers from being called' might be clearer.